### PR TITLE
chore(ember): update to ember 3.16

### PR DESCRIPTION
**Pull request by svsalvador**
BREAKING CHANGE: this update may break your app.

### CODE REVIEW

**Review by xe21500 — DISMISSED** [review]

File: addon/mixins/resolver.js [hunk 1]
```diff
@@ -469,13 +481,8 @@ export default Mixin.create({
 	 * @return String
 	 */
 	findModule(namespace, type, name) {
-		const str = makeToString(namespace, type, name);
 		const resolvers = A([this.findModuleByMain, this.findModuleByType, this.findModuleByPod]);
 
-		if (this._moduleCache[str]) {
```
Comment (adrigzr) [inline addon/mixins/resolver.js:475]: Why did you remove this? It is meant to be a cache to prevent resolve same modules again.

**Review by adrigzr — DISMISSED** [review]

File: addon/mixins/resolver.js [hunk 2]
```diff
@@ -196,6 +193,7 @@ function reopenModule(module) {
 		if (matches && namespaces.includes(matches[1])) {
 			const reopen = this._loadModule(entry);
 
+			/* istanbul ignore else */
```
Comment (josex2r) [inline addon/mixins/resolver.js]: Lower the coverage threshold in `package.json` instead of adding comments. It would be easy to forget that this addon does not accomplish the min requirements if the report always says that is covered 100%.

**Review by josex2r — CHANGES_REQUESTED** [review]

File: addon/mixins/resolver.js (hunk 2 above)
Comment (svsalvador) [reply addon/mixins/resolver.js]: You are right

File: addon/mixins/resolver.js (hunk 1 above)
Comment (svsalvador) [reply addon/mixins/resolver.js:475]: The Ember registry always answer  before the resolver:
https://github.com/emberjs/ember.js/blob/master/packages/%40ember/-internals/container/lib/registry.ts#L789

**Review by josex2r — APPROVED** [review]

**Review by eumendoza — APPROVED** [review]

File: addon/mixins/resolver.js (hunk 2 above)
Comment (kcmr) [reply addon/mixins/resolver.js]: I think it's not a bad idea to explicitly mark a part of the code as not being tested using this kind of comments while keeping the existing config for coverage thresholds. This way the requirements (100% coverage) and the parts not tested are clear.
